flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -412,28 +436,141 @@ public ScheduleMode getScheduleMode() {
 		return scheduleMode;
 	}
 
+	/**
+	 * Scheduling to be performed when an ExecutionVertex is encountered that cannot be resumed
+	 */
+	private interface ScheduleAction {
```

Comment:
Absolutely right. I was going to change their location before merging. They will probably go into a new class dedicated for the backtracking alongside with the `scheduleUsingBacktracking` method. I had trouble finding the right spot and decided to leave them there for now.